### PR TITLE
Zy/212 gui r scripts finish popup

### LINKED ISSUE #212: GUI: After R scripts finish popup a SnackBar to say the plot generation is completed.
Currently there is no indication when running scripts finishes. For example the generation of the Visual plots. It would be good to popup a SnackBar message when finished.

### PR DESCRIPTION
**Pull request by zheyxu**
# Pull Request Details

## What issue does this PR address

- GUI: After R scripts finish popup a SnackBar to say the plot generation is completed.

- Link to associated issue: #212 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added a reviewer

## Finalising

Once PR discussion is complete and reviewer has approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

File: assets/r/explore_summary.R [hunk 1]
```diff
@@ -75,3 +75,6 @@ timeDate::skewness(ds[numc], na.rm=TRUE)
 
 ## ") 
 ## }
+
+# Output a message when execution is complete
+print("Execution Completed")
```
Comment (gjwgit) [inline assets/r/explore_summary.R]: Please fix up the end of file - missing new line.

File: lib/r/source.dart [hunk 2]
```diff
@@ -323,4 +323,14 @@ Future<void> rSource(BuildContext context, WidgetRef ref, String script) async {
   code = rStripComments(code);
 
   ref.read(ptyProvider).write(const Utf8Encoder().convert(code));
+
+  // Optionally, show a SnackBar when the script starts executing.
```
Comment (gjwgit) [inline lib/r/source.dart]: Should this be showing the message when the script finished, not when it starts?

**Review by gjwgit — CHANGES_REQUESTED** [review]
Not sure I like this solution. I need to modify every script I write and that is a maintenance nightmare.

**Review by Lutra-Fs — COMMENTED** [review]
As we are currently facing the execution order problem, should we better include which file/function are completed?

**Comment by zheyxu** [comment]
> As we are currently facing the execution order problem, should we better include which file/function are completed?
Included:
Screenshot 2024-08-31 at 6 34 38 PM

**Comment by gjwgit** [comment]
Closing this PR as the solution is not efficient. The related issue moved back to In Progress. This might also be relevant to the async RScript redesign.

**Pull request closed without merging**